### PR TITLE
Refactor: replace OLAPStatus with Status in Tablet

### PR DESCRIPTION
**Pull request by sduzh**
Reference issue #60

### CODE REVIEW

File: be/src/storage/vectorized/tablet_reader.cpp [hunk 1]
```diff
@@ -33,14 +33,13 @@ void TabletReader::close() {
...
     _tablet->obtain_header_rdlock();
-    auto res = _tablet->capture_consistent_rowsets(_version, &_rowsets);
+    auto st = _tablet->capture_consistent_rowsets(_version, &_rowsets);
     _tablet->release_header_lock();
-    if (res == OLAP_SUCCESS) {
-        return Status::OK();
-    } else {
-        return Status::InternalError(
+    if (!st.ok()) {
+        st = Status::InternalError(
```
Comment (imay) [inline be/src/storage/vectorized/tablet_reader.cpp:39]: 1. Why not return st directly?
2. If return other status, it's better to print st.message()

File: be/src/storage/task/engine_clone_task.cpp [hunk 2]
```diff
@@ -618,24 +618,23 @@ Status EngineCloneTask::_clone_full_data(Tablet* tablet, TabletMeta* cloned_tabl
...
         }
-        s = rowset_to_remove->remove();
-        if (s != OLAP_SUCCESS) {
-            LOG(WARNING) << "failed to remove rowset " << rs_meta_ptr->rowset_id().to_string() << ", res=" << s;
+        if (auto ost = rowset_to_remove->remove(); ost != OLAP_SUCCESS) {
+            LOG(WARNING) << "failed to remove rowset " << rs_meta_ptr->rowset_id().to_string() << ", res=" << ost;
         }
     }
-    return ost == OLAP_SUCCESS ? Status::OK() : Status::InternalError("fail to revise tablet meta");
+    return st.ok() ? Status::OK() : Status::InternalError("fail to revise tablet meta");
```
Comment (imay) [inline be/src/storage/task/engine_clone_task.cpp:637]: return st?

**Review by kangkaisen — DISMISSED** [review]

File: be/src/storage/vectorized/tablet_reader.cpp (hunk 1 above)
Comment (sduzh) [reply be/src/storage/vectorized/tablet_reader.cpp]: updated

File: be/src/storage/task/engine_clone_task.cpp (hunk 2 above)
Comment (sduzh) [reply be/src/storage/task/engine_clone_task.cpp]: updated

**Review by kangkaisen — APPROVED** [review]

**Review by imay — APPROVED** [review]

**Pull request merged**